SonarQube.TeamBuild.PreProcessor/BuildWrapperInstaller.cs

```diff
@@ -0,0 +1,44 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="BuildWrapperInstaller.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using SonarQube.Common;
+using System;
+
+namespace SonarQube.TeamBuild.PreProcessor
+{
+    public class BuildWrapperInstaller : IBuildWrapperInstaller
```

Comment:
The implementation follows the same pattern as for the targets installer and analyzer installer: there is a separate component that handles the build wrapper code, which is created by the pre-processor object factory. The pre-processor just calls the factory to get the component and asks it to execute.